File: frontend/src/types/template.ts
export interface Template {
  id: string
  name: string
  description?: string
  category: string
  settings?: {
    general?: {
      isPublic?: boolean;
    };
    theme?: {
      chat?: {
        hostBubbles?: {
          backgroundColor?: string;
          color?: string;
        };
      };
    };
  }
} 

File: backend/src/main/java/io/typebot/features/template/model/Template.java
package io.typebot.features.template.model;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "templates")
public class Template {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    
    private String name;
    private String description;
    private String category;
    
    @Column(columnDefinition = "jsonb")
    private String settings;
    
    private boolean isPublic;
} 
